src/pokenav.c

```diff
@@ -0,0 +1,1198 @@
+        {
+            u32 arg = GetWordTaskArg((u8)i, 1);
+            if (arg == a0)
+                return TRUE;
+        }
+    }
+    return FALSE;
+}
+
+void sub_81C7170(u8 taskId)
+{
+    s16 *dataPtr;
+    u32 (*func)(u32);
+    bool32 exitLoop;
+
+    func = (u32 (*)(u32))GetWordTaskArg(taskId, 1);
+    dataPtr = gTasks[taskId].data;
+    exitLoop = FALSE;
+    while (!exitLoop)
+    {
+        u32 v1 =((u32 (*)(u32))func)(dataPtr[0]); 
+        switch (v1)
+        {
+        case 1:
+            dataPtr[0] = dataPtr[0] + 1;
+            break;
+        case 0:
+            dataPtr[0]++;
+            return;
+        case 4:
+            DestroyTask(taskId);
+            return;
+        default:
+            dataPtr[0] = v1 - 5;
+            break;
+        case 3:
+            break;
+        case 2:
+            return;
+        }
+    }
+}
+
+void sub_81C71E4(u8 taskId)
+{
+    u32 (*func)(u32);
+    s16 *data;
+    u32 v1;
+
+    if (sub_8087598())
+        return;
+    func = (u32 (*)(u32))GetWordTaskArg(taskId, 1);
+    data = gTasks[taskId].data;
+    v1 = func(data[0]);
+    switch (v1)
+    {
+    case 0:
+    case 1:
+        data[0]++;
+        break;
+    case 4:
+        DestroyTask(taskId);
+        break;
+    default:
+        data[0] = v1 - 5;
+        break;
+    case 2:
+    case 3:
+    }
+}
+
+void CB2_PokeNav(void)
+{
+    gUnknown_0203CF40 = Alloc(sizeof(struct UnknownStruct_0203CF40));
+    if (gUnknown_0203CF40 == NULL)
+        SetMainCallback2(CB2_ReturnToFieldWithOpenMenu);
+    else
+    {
+        sub_81C7360(gUnknown_0203CF40);
+        ResetTasks();
+        SetVBlankCallback(NULL);
+        CreateTask(sub_81C742C, 0);
+        SetMainCallback2(sub_81C7400);
+        SetVBlankCallback(sub_81C7418);
+    }
+}
+
+void sub_81C72A4()
+{
+    SetMainCallback2(sub_81C72BC);
+    FadeScreen(1, 0);
+}
+
+void sub_81C72BC()
+{
+    UpdatePaletteFade();
+    if (!gPaletteFade.active)
+    {
+        gUnknown_0203CF40 = Alloc(sizeof(struct UnknownStruct_0203CF40));
+        if (gUnknown_0203CF40 == NULL)
+            SetMainCallback2(CB2_ReturnToFieldContinueScriptPlayMapMusic);
+        else
+        {
+            sub_81C7360(gUnknown_0203CF40);
+            gUnknown_0203CF40->field8 = 1;
+            ResetTasks();
+            ResetSpriteData();
+            FreeAllSpritePalettes();
+            SetVBlankCallback(NULL);
+            CreateTask(sub_81C742C, 0);
+            SetMainCallback2(sub_81C7400);
+            SetVBlankCallback(sub_81C7418);
+        }
+    }
+}
+
+void sub_81C7334()
+{
+    s32 i;
+    for (i = 0; i < 19; i++)
+    {
+        sub_81C7650(i);
+    }
+
+    Free(gUnknown_0203CF40);
+    gUnknown_0203CF40 = NULL;
+    InitKeys();
+}
+
+// Clears UnknownStruct_0203CF40
+void sub_81C7360(struct UnknownStruct_0203CF40 *a0)
+{
+    s32 i;
+    void **arrayPtr;
+    void *fill;
+
+    fill = NULL;
+    i = 18;
+    arrayPtr = &(a0->field10[18]);
+    for (i; i >= 0; i--)
```

Comment:
Sure this isn't a compiler-optimized memset?

Reply:
`memset(&a0->field10, 0, 19);` failed to replace it, so I don't think so.